cmake/sdlcompilers.cmake

```diff
@@ -0,0 +1,144 @@
+macro(SDL_DetectCompiler)
+  set(USE_CLANG FALSE)
+  set(USE_GCC FALSE)
+  set(USE_INTELCC FALSE)
+  set(USE_QCC FALSE)
+  if(CMAKE_C_COMPILER_ID MATCHES "Clang|IntelLLVM")
+    set(USE_CLANG TRUE)
+    # Visual Studio 2019 v16.2 added support for Clang/LLVM.
+    # Check if a Visual Studio project is being generated with the Clang toolset.
+    if(MSVC)
+      set(MSVC_CLANG TRUE)
+    endif()
```

Comment:
This is the condition for `MSVC_CLANG`.

Reply:
That is indeed clang-cl.

Reply:
> That is indeed clang-cl.

I mean: arm32 linkage is failing most probably because you set the condition for MSVC_CLANG, but the compiler is simply MSVC without clang.

Reply:
It was failing because the architecture was not detected. Now it does.